File: api/src/main/java/net/asch/bulkit/api/capability/IDiskResourceHandler.java
package net.asch.bulkit.api.capability;

import net.neoforged.neoforge.items.IItemHandler;

import javax.annotation.Nullable;

public interface IDiskResourceHandler {
    long getAmount();
    void setAmount(long value);
    boolean isLocked();
    void setLocked(boolean value);
    boolean isVoidExcess();
    void setVoidExcess(boolean value);
    @Nullable IItemHandler mods();

    int multiplier(int defaultMultiplier);
}

File: api/src/main/kotlin/net/asch/bulkit/api/capability/IDiskResourceHandler.kt
package net.asch.bulkit.api.capability

import net.neoforged.neoforge.items.IItemHandler

interface IDiskResourceHandler {
    var amountI: Int
    var amountL: Long

    var isLocked: Boolean
    var isVoidExcess: Boolean
    val mods: IItemHandler?

    fun getMultiplier(defaultMultiplier: Int): Int
}